backend/src/main/java/gov/cdc/usds/simplereport/db/repository/AdvisoryLockManager.java

```diff
@@ -0,0 +1,39 @@
+package gov.cdc.usds.simplereport.db.repository;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.query.Procedure;
+
+/**
+ * A mix-in repository for entity repositories that manage processes that use
+ * advisory locks.
+ */
+public interface AdvisoryLockManager {
+
+    int CORE_API_LOCK_SCOPE = 110506458; // some arbitrary 32-bit number that defines "our" locks
+    /**
+     * Take the advisory lock defined by the two arguments, waiting until the lock
+     * is available and releasing it at the end of the current transaction.
+     *
+     * @param lockCategory
```

Comment:
IDE wants description here, too e.g. `@param lockCategory  description goes here`

Reply:
Solid point, especially since there's a bit of ad-hoc structure being imposed here.